src/overlays/actors/ovl_Bg_Crace_Movebg/z_bg_crace_movebg.c

```diff
@@ -44,136 +51,151 @@ const ActorInit Bg_Crace_Movebg_InitVars = {
     (ActorFunc)BgCraceMovebg_Draw,
 };
 
-static u8 D_80A710A0 = 0;
-static u8 D_80A710A4 = 0;
+static u8 sHasInitializedIsLoaded = 0;
+static u8 sLoadedDoorCount = 0;
 static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
 };
-static Vec3f D_80A710AC[] = { 0.0f, 0.0f, 1.0f };
-u8 D_80A711B0[32];
+static Vec3f sUnitVecZ = { 0.0f, 0.0f, 1.0f };
+u8 sIsLoaded[32];
 
 void BgCraceMovebg_Init(Actor* thisx, PlayState* play) {
     BgCraceMovebg* this = THIS;
     s32 i;
     s32 j;
 
     Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
-    if (D_80A710A0 == 0) {
-        for (i = 0; i < ARRAY_COUNT(D_80A711B0); i++) {
-            D_80A711B0[i] = 0;
+    if (!sHasInitializedIsLoaded) {
+        for (i = 0; i < ARRAY_COUNT(sIsLoaded); i++) {
+            sIsLoaded[i] = 0;
         }
-        D_80A710A0 = 1;
+        sHasInitializedIsLoaded = true;
     }
 
     DynaPolyActor_Init(&this->dyna, 1);
-    DynaPolyActor_LoadMesh(play, &this->dyna, &object_crace_object_Colheader_000E00);
+    DynaPolyActor_LoadMesh(play, &this->dyna, &gDekuShrineSlidingDoorCol);
 
-    this->unk184 = this->dyna.actor.world.rot.x;
-    this->unk16C = this->dyna.actor.world.rot.z;
+    this->index = BG_CRACE_MOVEBG_GET_INDEX(&this->dyna.actor);
+    this->openSpeed = BG_CRACE_MOVEBG_GET_OPEN_SPEED(&this->dyna.actor);
     this->dyna.actor.world.rot.x = 0;
     this->dyna.actor.world.rot.z = 0;
     this->dyna.actor.home.rot.x = 0;
     this->dyna.actor.home.rot.z = 0;
     this->dyna.actor.flags |= ACTOR_FLAG_10000000;
 
-    switch (BGCRACEMOVEBG_GET_F(&this->dyna.actor)) {
-        case 0:
-        case 2:
-            for (j = 0; j < D_80A710A4; j++) {
-                if (D_80A711B0[j] == this->unk184) {
-                    this->unk170 |= 8;
+    switch (BG_CRACE_MOVEBG_GET_TYPE(&this->dyna.actor)) {
+        case BG_CRACE_MOVEBG_TYPE_CLOSING:
+        case BG_CRACE_MOVEBG_TYPE_UNUSED_CLOSING:
+            for (j = 0; j < sLoadedDoorCount; j++) {
+                if (sIsLoaded[j] == this->index) {
+                    this->flags |= BG_CRACE_MOVEBG_FLAG_ALREADY_LOADED;
                     Actor_MarkForDeath(&this->dyna.actor);
                     return;
                 }
             }
-            if (D_80A710A4 < ARRAY_COUNT(D_80A711B0)) {
-                D_80A711B0[(s32)D_80A710A4++] = this->unk184;
+
+            if (sLoadedDoorCount < ARRAY_COUNT(sIsLoaded)) {
+                sIsLoaded[(s32)sLoadedDoorCount++] = this->index;
```

Comment:
I suppose it's required to write it like this, the increment can't be separate?

Reply:
The increment can actually be separate. The (s32) cast is seemingly required.